src/commands/registerCommands.ts

```diff
@@ -139,49 +139,93 @@ for (const label of bindingTypes.keys()) {
+    canSelectMany: false,
+    openLabel: 'Select XSD or DTD file',
+    filters: {
+      'Grammar files': ['xsd', 'dtd']
     }
-    const bindingType = bindingTypes.get(pickedBindingTypeOption.label);
-
-    // Open a dialog to select the XSD, DTD to bind.
-    const options: OpenDialogOptions = {
-      canSelectMany: false,
-      openLabel: 'Select XSD or DTD file',
-      filters: {
-        'Grammar files': ['xsd', 'dtd']
+  };
+
+  const fileUri = await window.showOpenDialog(options);
+  if (fileUri && fileUri[0]) {
+    // The XSD, DTD has been selected, get the proper syntax for binding this grammar file in the XML document.
+    const identifier = TextDocumentIdentifier.create(documentURI.toString());
+    const grammarURI = fileUri[0];
+    try {
+      const result = await commands.executeCommand(ServerCommandConstants.ASSOCIATE_GRAMMAR_INSERT, identifier, grammarURI.toString(), bindingType);
+      // Insert the proper syntax for binding
+      const lspTextDocumentEdit = <TextDocumentEdit>result;
+      const workEdits = new WorkspaceEdit();
+      for (const edit of lspTextDocumentEdit.edits) {
+        workEdits.replace(documentURI, languageClient.protocol2CodeConverter.asRange(edit.range), edit.newText);
       }
+      workspace.applyEdit(workEdits); // apply the edits
+
+    } catch (error) {
+      window.showErrorMessage('Error during grammar binding: ' + error.message);
     };
+  }
+}
 
-    const fileUri = await window.showOpenDialog(options);
-    if (fileUri && fileUri[0]) {
-      // The XSD, DTD has been selected, get the proper syntax for binding this grammar file in the XML document.
-      const identifier = TextDocumentIdentifier.create(documentURI.toString());
-      const grammarURI = fileUri[0];
-      try {
-        const result = await commands.executeCommand(ServerCommandConstants.ASSOCIATE_GRAMMAR_INSERT, identifier, grammarURI.toString(), bindingType);
-        // Insert the proper syntax for binding
-        const lspTextDocumentEdit = <TextDocumentEdit>result;
-        const workEdits = new WorkspaceEdit();
-        for (const edit of lspTextDocumentEdit.edits) {
-          workEdits.replace(documentURI, languageClient.protocol2CodeConverter.asRange(edit.range), edit.newText);
-        }
-        workspace.applyEdit(workEdits); // apply the edits
-      } catch (error) {
-        window.showErrorMessage('Error during grammar binding: ' + error.message);
-      };
+/**
+ * Register commands used for associating grammar file (XSD,DTD) to a given XML file for command menu and CodeLens
+ *
+ * @param context the extension context
+ * @param languageClient the language server client
+ */
+ function registerAssociationCommands(context: ExtensionContext, languageClient: LanguageClient) {
+  // For CodeLens
+  context.subscriptions.push(commands.registerCommand(ClientCommandConstants.OPEN_BINDING_WIZARD, async (uri: Uri) => {
+    grammarAssociationCommand(uri, languageClient)
```

Comment:
The parameter uri is a string for codelens. Déclare parameter with uriString:string and convertir it 
const uri= Uri.parse(uriString);